cpp/src/arrow/array/data.h

```diff
@@ -449,7 +449,8 @@ struct ARROW_EXPORT ArraySpan {
   /// \return A span<const T> of the requested length
   template <typename T>
   util::span<const T> GetSpan(int i, int64_t length) const {
-    const int64_t buffer_length = buffers[i].size / static_cast<int64_t>(sizeof(T));
+    [[maybe_unused]] const int64_t buffer_length =
```

Comment:
can we just remove this line?

cc @jsjtxietian

Reply:
Oh we cannot, `assert` will check it when debug mode

Reply:
`ARROW_UNUSED()` looks better

Reply:
> `ARROW_UNUSED()` looks better

Will replace it with `ARROW_UNUSED()`. Thanks.

Reply:
> can we just remove this line?
> 
> cc @jsjtxietian

It's quite an important assert, because the caller is supposed to pass the correct length when calling these functions.